integration_tests/integration_test.py

```diff
@@ -93,25 +71,38 @@ def setUp(self):
         self._test_success = True
 
     def tearDown(self):
-        assert self.container_id
-        execute('docker', 'stop', self.container_id)
-        execute('docker', 'rm', self.container_id)
-
+        super().tearDown()
         self._reset_logging()
 
-        cfg.content['jobs_dir'] = self.original_job_dir
-        cfg.content['run']['input_dir'] = self.original_run_input
-        cfg.content['run']['output_dir'] = self.original_run_output
-        cfg.content['sample']['output_dir'] = self.original_sample_output
+    def setup_test(self, test_type, test_name, integration_section, species='Homo sapiens', analysis_type='Variant Calling gatk'):
+        cfg.content['jobs_dir'] = os.path.join(os.path.dirname(os.getcwd()), 'jobs', test_name)
+        cfg.content[test_type]['output_dir'] = os.path.join(os.getcwd(), 'outputs', test_name)
+        if 'input_dir' in self.cfg[integration_section]:
+            cfg.content[test_type]['input_dir'] = self.cfg[integration_section]['input_dir']
+
+        os.makedirs(cfg['jobs_dir'])
+        os.makedirs(cfg[test_type]['output_dir'])
+
+        def _fake_get_sample(sample_name):
+            return Mock(
+                name=sample_name,
+                udf={
+                    'Coverage': 1337,
+                    'Analysis Type': analysis_type,
+                    'Yield for Quoted Coverage (Gb)': 15,
+                    'Required Yield (Gb)': 30,
+                    'Coverage (X)': 15,
+                }
+            )
 
-    @staticmethod
-    def setup_test(test_type, test_name, integration_section):
-        cfg.content['jobs_dir'] = os.path.join(cfg['jobs_dir'], test_name)
-        cfg.content[test_type]['output_dir'] = os.path.join(cfg[test_type]['output_dir'], test_name)
-        if 'input_dir' in integration_cfg[integration_section]:
-            cfg.content[test_type]['input_dir'] = integration_cfg[integration_section]['input_dir']
-        os.mkdir(cfg['jobs_dir'])
-        os.mkdir(cfg[test_type]['output_dir'])
+        patches = (
+            patch('egcg_core.clarity.get_sample', new=_fake_get_sample),
+            patch('egcg_core.clarity.get_species_from_sample', return_value=species)
+        )
+        for p in patches:
+            p.start()
+
+        self.patches += patches
```

Comment:
This will not happen because a new IntegrationTest object is constructed per test method run. Can move these extra patches to their own attribute to make it more explicit